kpatch-build/create-diff-object.c

```diff
@@ -1555,7 +1561,8 @@ static void kpatch_replace_sections_syms(struct kpatch_elf *kelf)
 			    !strcmp(rela->sym->name, ".fixup") ||
 			    !strcmp(rela->sym->name, ".altinstr_replacement") ||
 			    !strcmp(rela->sym->name, ".altinstr_aux") ||
-			    !strcmp(rela->sym->name, ".text..refcount"))
+			    !strcmp(rela->sym->name, ".text..refcount") ||
+			    !strncmp(rela->sym->name, ".data..Lubsan_type", 18))
 				continue;
```

Comment:
When building a sample patch I got the following error
```
create-diff-object: ERROR: array.o: kpatch_replace_sections_syms: 1716: .text.do_task_stat+0xe0a: can't find replacement symbol for .data..Lubsan_data132+0 reference
```
so here I think we need something like this instead
```
			/*
			 * UBSAN data will be taken wholesale, no need to
			 * replace section symbols.
			 */
			if (is_ubsan_data(rela->sym->name))
				continue;

```

Reply:
When I looked at the kelf->symbols list,
sym->sec->sym->name: .Lubsan_data151 sym->sec->name: .data.rel.local..Lubsan_data151 sym->name: .data.rel.local..Lubsan_data151

Relocation section '.rela.text.meminfo_proc_show' at offset 0x1e18 contains 154 entries:
    Offset             Info             Type               Symbol's Value  Symbol's Name + Addend
0000000000000ed6  0000004f0000001a R_390_GOTENT           0000000000000000 .Lubsan_data150 + 2
0000000000000edc  0000009600000014 R_390_PLT32DBL         0000000000000000 __ubsan_handle_out_of_bounds + 2
**0000000000000ee6  0000004e0000001a R_390_GOTENT           0000000000000000 .Lubsan_data151 + 2  <----**
0000000000000eec  0000009600000014 R_390_PLT32DBL         0000000000000000 __ubsan_handle_out_of_bounds + 2

The above rela entry **.Lubsan_data151** already had section symbol, so it wasnt replaced in kpatch_replace_sections_syms.

But I assume if the compiler generated rela entry such that symbol name is .data.rel.local..Lubsan_data151 
in .rela.text.meminfo_proc_show, then it should be replaced to .Lubsan_data151+offset in .rela.text.meminfo_proc_show. 

So, to be sure, I just added only .data..Lubsan_type case in kpatch_replace_sections_syms().
I will also add .data..Lubsan_data in addition to that. As this was missed  as you reported above.

In any case, if you think, we should cover all the case, I can add is_ubsan_sec() in kpatch_replace_sections_syms().

Reply:
Added the changes you mentioned, Agree to what you mentioned above. Thanks